### PR TITLE
First Pass at Filamentphp V3 support.

### PR DESCRIPTION
**Pull request by tonypartridge**
This will require a new branch since we cannot have a v2 and v3 version. 

this is a first pass at a read-only version. Will do more testing and implementation with creation/editing at some point too, but allows people to get off the ground.

### CODE REVIEW

**Comment by ana-lisboa** [comment]
Any chance this could possibly be merged alongside the possibility to delete events?

File: src/FilamentFullCalendarServiceProvider.php [hunk 1]
```diff
@@ -2,21 +2,24 @@
...
     public static string $name = 'filament-fullcalendar';
 
-    protected array $beforeCoreScripts = [
-        'filament-fullcalendar-scripts' => __DIR__ . '/../dist/filament-fullcalendar.js',
-    ];
-
-    protected array $styles = [
-        'filament-fullcalendar-styles' => __DIR__ . '/../dist/filament-fullcalendar.css',
-    ];
+    public function packageRegistered(): void
```
Comment (ziming) [inline src/FilamentFullCalendarServiceProvider.php:16]:
```suggestion
    public function packageBooted(): void
```

Asset registration should go into the packageBooted method

See:

https://filamentphp.com/docs/3.x/support/plugins/build-a-panel-plugin

https://filamentphp.com/docs/3.x/support/plugins/build-a-standalone-plugin

**Review by ziming — CHANGES_REQUESTED** [review]

**Comment by lam0819** [comment]
your version quite few things need to change

Here is my version of upgrade v3, I have using in my project now (as local package)
https://github.com/saade/filament-fullcalendar/pull/99

**Comment by saade** [comment]
Hello friends.

Just released v3.0.0-beta1.

PLEASE READ CAREFULLY:

- This is a first-pass release for supporting v3.
- There WILL BE BRAKING CHANGES when the final v3.0.0 is released.
- The docs are not updated yet. But there are no braking changes for now.


Thanks for everyone who opened a PR for supporting v3. :yellow_heart:

**Pull request closed without merging**